design/secrets.md

```diff
@@ -84,6 +147,22 @@ Currently, GCP is the only provider that relies on the existing environment vari
 For GCP, the environment variable will be overwritten with the path of the serialized secret.
 
 
+## Split credentials between VolumeSnapshotter and ObjectStore plugins
+
+One of the use cases we wish to satisfy is the ability to specify a different object store than the cloud provider offers,
+for example, using a Minio S3 object store from within AWS.  Currently the VolumeSnapshotter and the ObjectStore plugin
+share the cloud credentials.  The VolumeSnapshotter cannot use the Minio credentials to take snapshots.  We need to split
+the VolumeSnapshotter credentials from the ObjectStore credentials.  We will do this by always using the default cloud
+credentials with the VolumeSnapshotter plugins and not passing the BSL credentials in the current implementation.
+
+## Velero Plugin for vSphere compatibility
+
+The vSphere plugin is implemented as a BackupItemAction and shares the credentials of the AWS plug-in for S3 access.
+The backup storage location is passed in _Backup.Spec.StorageLocation_.  Currently the plugin retrieves the S3 bucket and
+server from the BSL and creates a BackupRespositoryClaim with that and the credentials retrieved from the cloud credential.
+The plug-in will need to be modified to retrieve the credentials field from the BSL and use that credential in the
```

Comment:
Having looked through the VSphere plugin code, it seems that [this is the section that will need to change](https://github.com/vmware-tanzu/velero-plugin-for-vsphere/blob/d2cba53d3b8092ac8562df025688daac06e9bd16/pkg/utils/utils.go#L172-L209) as it's currently using the fixed secret name and just selecting the first data field within that secret.

The current design is passing through a file path which has the credential written to disk which won't work. I haven't looked at the plugin code in depth but it seems like it may be enough to also include the secret name and key in the config as well. This would allow the VSphere plugin to use those values from the BSL config to retrieve the necessary secret to configure the AWS client. This effectively combines options 2 and 3.